include/dali/core/cuda_stream.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_CUDA_STREAM_H_
+#define DALI_CORE_CUDA_STREAM_H_
+
+#include <driver_types.h>
+#include <utility>
+#include "dali/core/unique_handle.h"
+
+namespace dali {
+
+class DLL_PUBLIC CUDAStream : public UniqueHandle<cudaStream_t, CUDAStream>{
```

Comment:
I know that it's something we don't agree on, but there, in the [code style] (https://google.github.io/styleguide/cppguide.html#General_Naming_Rules) you explicitly have the naming convention as `CudaStream`.

Here I'd like to mention, that in my opinion it really is better choice, as you yourself came up to this in this particual PR, here:
```
class DLL_PUBLIC StftGPU {
```
You used `StftGPU`, not `STFTGPU` for this particual reason: acronyms shall have camel case. In my opinion we should stick to `CudaStream`, `StftGpu`, `XmlHttpRequest` etc...

Reply:
There are 10x more GPU than Gpu in our code - and that's a fair comparison (i.e. it includes only identifiers where the preceding or succeeding character is a lowercase letter) - including the major APIs like CPUBackend, GPUBackend, RunGPU, RunCPU.
It's not STFTGPU only because it would look like a macro. But since it already has all the capabilities to calculate a spectrogram, I can rename it accordingly.

Reply:
@mzient - again let us agree about exception if we think that this rule doesn't help.